src/objective/hinge.cu

```diff
@@ -75,31 +57,26 @@ class HingeObj : public ObjFunction {
           }
           _out_gpair[_idx] = GradientPair(g, h);
         },
-        common::Range{0, static_cast<int64_t>(ndata)}, devices_).Eval(
-            &label_correct_, out_gpair, &preds, &info.labels_, &info.weights_);
+        common::Range{0, static_cast<int64_t>(ndata)},
+        GPUSet::All(tparam_->gpu_id, tparam_->n_gpus, ndata)).Eval(
+            out_gpair, &preds, &info.labels_, &info.weights_);
```

Comment:
Note to myself: understand how `GPUSet::Eval()` works, so that we can implement ranking metrics (AUC/AUCPR/NDCG) on GPU in the future

Reply:
@hcho3 I have been reading up about that.  Thanks for your pointer to that paper.